store/file/ods.go

```diff
@@ -108,22 +108,45 @@ func writeODSFile(w io.Writer, eds *rsmt2d.ExtendedDataSquare, axisRoots *share.
 		return nil, fmt.Errorf("writing axis roots: %w", err)
 	}
 
-	for _, shr := range eds.FlattenedODS() {
-		if _, err := w.Write(shr); err != nil {
-			return nil, fmt.Errorf("writing shares: %w", err)
-		}
+	// write quadrants
+	err = writeQuadrant(w, eds, 0)
+	if err != nil {
+		return nil, fmt.Errorf("writing Q1: %w", err)
 	}
 	return h, nil
 }
 
+func writeQuadrant(w io.Writer, eds *rsmt2d.ExtendedDataSquare, quadrantIdx int) error {
+	buf := make([]byte, 0, share.Size*eds.Width())
```

Comment:
I think on the read side we would benefit a lot by reading row by row, rather than share by share. To be proven with benchmarks

Reply:
It this the case, there is even wip draft PR for that and some memory optimisations for reading buffer as well
https://github.com/celestiaorg/celestia-node/pull/3512